packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -38,6 +38,9 @@ export class EngineReportingExtension<TContext = any>
   private options: EngineReportingOptions<TContext>;
   private addTrace: (args: AddTraceArgs) => Promise<void>;
   private generateClientInfo: GenerateClientInfo<TContext>;
+  private shouldReportQuery: (
+    request: GraphQLRequestContext<TContext>,
+  ) => boolean;
```

Comment:
How about the addition of a shorthand of just `boolean` rather than `() => boolean` (e.g. `shouldReportQuery: false`)?